minari/utils/data_collector.py

```diff
@@ -0,0 +1,573 @@
+from __future__ import annotations
+
+import os
+import shutil
+import tempfile
+from typing import Any, Dict, Optional, Type, Union
+
+import gymnasium as gym
+import h5py
+import numpy as np
+from gymnasium import spaces
+
+
+STEP_DATA_KEYS = {
+    "actions",
+    "observations",
+    "rewards",
+    "truncations",
+    "terminations",
+}
+
+
+class EpisodeMetadataCallback:
+    """Callback to full episode after saving to hdf5 file as a group.
+
+    This callback can be overridden to add extra metada attributes or statistics to
+    each HDF5 episode group in the Minari dataset. The custom callback can then be
+    passed to the DataCollectorV0 wrapper in the `episode_metadata_callback` argument.
+
+    TODO: add more default statistics to episode datasets
```

Comment:
When will this be done?

Reply:
Haven't set a timeline for it. I have to investigate a bit more what would be useful, open to suggestions. This will probably be done in separate PR.